modules/javafx.base/src/main/java/javafx/beans/property/ReadOnlySetProperty.java

```diff
@@ -111,7 +111,10 @@ public boolean equals(Object obj) {
 
         if (!(obj instanceof Set))
             return false;
-        Set c = (Set) obj;
+
+        @SuppressWarnings("unchecked")
+        Set<E> c = (Set<E>) obj;  // safe cast as elements are only referenced
+
```

Comment:
NPE is probably because some underlying sets will throw NPE even for operations like `contains(null)` (the immutable variants from `List.of` and `Set.of` are bad offenders in this area).  CCE maybe for sets that use comparators internally.

I've adjusted it as you suggested, and same for map.